tests/containerTests.js

```diff
@@ -34,6 +34,15 @@ describe('Container', () =>  {
             expect(foo.bar).toBe(5);
         });
 
+        it('should register/resolve an object with a custom factory', () =>  {
+            let Foo = function(){};
```

Comment:
I think all you need for this test is `let Foo = createObject({ bar: { value : 5 }});`,  You don't really need to assign the `prototype`.